File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        int k = 1234567;
        if(n == 2){
            return 1;
        }
        
        int a = 0, b= 1, temp;
        
        for(int i = 2; i <= n; i++){
            temp = (a+b) % k;
            a = b;
            b = temp;
        }
        
        return b;
    }
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.js
function solution(n) {
    var answer = 0;
    let divider = 1234567;
    let pivot = new Array(n+1);
    
    pivot[0] = 0;
    pivot[1] = 1;
    
    for(var i = 2; i <= n; i++){
        pivot[i] = (pivot[i-2] + pivot[i-1])%divider;
    }
    return pivot[n];
}